### ISSUE TITLE
Global ResolvedTs should be reset to CheckpointTs when a capture fails

### ISSUE THREAD

**Issue by liuzix** [pingcap/ticdc#860]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
I made a capture panic by adding some unrelated code. (The first RowChangedEvent of the changefeed made the sink panic)

2. What did you expect to see?
The capture should be recreated on some node and try to sync the RowChangedEvent again. 

3. What did you see instead?
Any restarted capture will panic because it tries to pull starting from the global CheckpointTs, which is smaller than the global ResolvedTs. Error info: "**The CRTs of event is not expected, please report a bug**".

4. Versions of the cluster
    - TiCDC version (execute `cdc version`):

        ```
        Forked from master 9f21207123d03de3ee7c056f92397a5b2bd0a1cc
        ```

**Comment by amyangfei** [comment]
Could you please add detail panic log

**Comment by amyangfei** [comment]
fixed by #875
